feat: allow editing completion date

creightonfrance opened the pull request:
- #1184 

This PR will allow changing the task completion date. 

- If user tries to set completion date in the future, they will see error dialog.
- If user deletes completion date, the task will marked as not completed.

raimund-schluessler commented:
Seems to work fine. But I am not sure we should show this date so prominently. I would rather move it down into the details section. Probably below the `status` drop-down. What do you think?

creightonfrance commented:
That makes sense, it might be better under the status input since that determines this field's visibility. Making that change now 👍🏻

creightonfrance commented:
# Changes
I moved the field down to the details section. I also added a flag for the overdue check.

# Question
Could you help me out with local lint checks? I run `make` and  `npm run lint`, and I get hundreds of errors (mainly issues finding dependencies). I've fully opened up permissions locally for all files via `chmod 777 * -R`. Am I missing a step?

raimund-schluessler commented:
> # Question
> 
> Could you help me out with local lint checks? I run `make` and `npm run lint`, and I get hundreds of errors (mainly issues finding dependencies). I've fully opened up permissions locally for all files via `chmod 777 * -R`. Am I missing a step?

Unfortunately, I see the same locally. And I couldn't figure out yet, why that happens. Probably worth opening an issue.

raimund-schluessler commented on src/components/AppSidebar/DateTimePickerItem.vue:
```diff
@@ -115,6 +115,13 @@ export default {
+		 */
+		checkOverdue: {
+			type: Boolean,
+			default: false,
```
Could you please make the default `true` to keep the behavior the same?

creightonfrance commented:
> > # Question
> > Could you help me out with local lint checks? I run `make` and `npm run lint`, and I get hundreds of errors (mainly issues finding dependencies). I've fully opened up permissions locally for all files via `chmod 777 * -R`. Am I missing a step?
> 
> Unfortunately, I see the same locally. And I couldn't figure out yet, why that happens. Probably worth opening an issue.

I figured it out: issue only happens when the tasks repo is under the nextcloud/server repo (in apps, extra-apps, etc.). You have to run `make` in the base directory for nextcloud server, then `make` for this repo and lint should work fine 👍🏻 no code needed

raimund-schluessler reviewed (approved):
Seems to work fine. And sorry it took me so long to have a look.
